### ISSUE TITLE
scala.MatchError occurs when parsing a URL containing an IP address

### ISSUE THREAD

**Issue by garbagetown** [amuramatsu/gitbucket-rst-plugin#4]
scala.MatchError occurs when parsing a URL containing an IP address like below.

```
http://10.20.30.40/
```

The stacktrace is as follows.

```
scala.MatchError: 10 (of class java.lang.Integer)
	at laika.parse.util.URIParsers.flatten(URIParsers.scala:40)
	at laika.parse.util.URIParsers.flatten$(URIParsers.scala:34)
	at laika.parse.rst.ReStructuredText$StrictParsers$1.flatten(ReStructuredText.scala:237)
	at laika.parse.util.URIParsers.flatten(URIParsers.scala:40)
	at laika.parse.util.URIParsers.flatten$(URIParsers.scala:34)
	at laika.parse.rst.ReStructuredText$StrictParsers$1.flatten(ReStructuredText.scala:237)
	at laika.parse.util.URIParsers.$anonfun$ipv4address$5(URIParsers.scala:125)
	at scala.util.parsing.combinator.Parsers$Success.map(Parsers.scala:137)
```

I know that this error is not your amazing plugin's bug, but the LAIKA parser's bug. I think it would be better to show the raw ReST text when parse error occurs. Thanks.

**Comment by amuramatsu** [comment]
I checked this issue.

This is bug of ReST rendering library as you said, but this plugin must not crash.
I'll fix to catch all exceptions in plugin, and show raw ReST and Exceptions as rendering result when exception is occured.

Also I'll report this bug to author of rendering library.

**Comment by amuramatsu** [comment]
Crash problem is fixed at version 0.3.0.

Reporting to LAIKA is not finished yet.

**Comment by garbagetown** [comment]
I've confirmed now the plugin does not crash when parse errors occur! Great thanks!!

**Comment by amuramatsu** [comment]
I reported this issue to LAIKA author (https://github.com/planet42/Laika/issues/45).

**Comment by amuramatsu** [comment]
fixed by commit f9f737aad239e660452d9a4f6a824ccb2c48fc5c
